eclair-core/src/main/scala/fr/acinq/eclair/payment/receive/MultiPartHandler.scala

```diff
@@ -251,22 +250,42 @@ object MultiPartHandler {
                                     paymentPreimage_opt: Option[ByteVector32] = None,
                                     paymentType: String = PaymentType.Standard) extends ReceivePayment
 
+  /**
+   * A dummy blinded hop that will be added at the end of a blinded route.
+   * The fees and expiry delta should match those of real channels, otherwise it will be obvious that dummy hops are used.
+   */
+  case class DummyBlindedHop(feeBase: MilliSatoshi, feeProportionalMillionths: Long, cltvExpiryDelta: CltvExpiryDelta)
```

Comment:
These fees and expiry delta won't be visible, only the aggregated value with the other hops. It seems weird to allow the user to choose where to put the fees when the results will be indistinguishable.
You can have a dummy hop with fees followed by a dummy hop without fees, the same in reverse order, or just two hops with half the fees each and the resulting blinded routes will look exactly the same to a third party.

Reply:
> These fees and expiry delta won't be visible, only the aggregated value with the other hops.

But if they're set to 0, the sender can trivially figure out that we're using only dummy hops since the aggregated values will be very low. It feels to me that a smart(ish) way of generating dummy hops is by looking at all credible hops going out and selecting fees/expiry that match most of these real hops, and iterating like that, that's why I made it that way. It also forces the caller to think about the fact that _each_ dummy hop should add real fees/expiry to be useful to avoid reducing the anonymity set.